packages/common/src/pivot/derivePivotConfigFromChart.ts

```diff
@@ -0,0 +1,231 @@
+        reference: col,
+    }));
+
+    const partialPivotConfiguration: Omit<PivotConfiguration, 'sortBy'> = {
+        indexColumn,
+        valuesColumns,
+        groupByColumns,
+    };
+
+    const pivotConfiguration: PivotConfiguration = {
+        ...partialPivotConfiguration,
+        sortBy: getSortByForPivotConfiguration(
+            partialPivotConfiguration,
+            metricQuery,
+        ),
+    };
+
+    return pivotConfiguration;
+}
+
+function getCartesianPivotConfiguration(
+    savedChart: SavedChartDAO,
+    metricQuery: MetricQuery,
+    fields: ItemsMap,
+): PivotConfiguration | undefined {
+    const { chartConfig } = savedChart;
+
+    if (chartConfig.type !== ChartType.CARTESIAN) {
+        throw new Error('Chart is not a Cartesian chart');
+    }
+
+    if (!isCartesianChartConfig(chartConfig.config)) {
+        throw new Error('Invalid cartesian chart config - no eCharts config');
+    }
+
+    const { eChartsConfig } = chartConfig.config;
+    const series = eChartsConfig?.series;
+    const firstSeries = series?.[0];
+
+    if (firstSeries && firstSeries.encode.yRef.pivotValues?.length) {
+        // Extract pivot configuration from the first series
+        const { yRef, xRef } = firstSeries.encode;
+
+        if (yRef.pivotValues) {
+            // Extract unique group by columns from pivot values
+            const groupByColumns = yRef.pivotValues.map((pv: PivotValue) => ({
+                reference: pv.field,
+            }));
+
+            // Extract value columns
+            const valuesColumns = uniqBy(series, 'encode.yRef.field').map(
+                (s: Series) => ({
+                    reference: s.encode.yRef.field,
+                    aggregation: VizAggregationOptions.ANY,
+                }),
+            );
+
+            const xAxisDimension = fields[xRef.field];
+            let xAxisType: VizIndexType | undefined;
+
+            if (xAxisDimension && isDimension(xAxisDimension)) {
+                xAxisType = getColumnAxisType(xAxisDimension.type);
+            }
+
+            const indexColumn = xAxisType
+                ? {
+                      reference: xRef.field,
+                      type: xAxisType,
+                  }
+                : undefined;
+
+            const partialPivotConfiguration: Omit<
+                PivotConfiguration,
+                'sortBy'
+            > = {
+                indexColumn,
+                valuesColumns,
+                groupByColumns,
+            };
+
+            const pivotConfiguration: PivotConfiguration = {
+                ...partialPivotConfiguration,
+                sortBy: getSortByForPivotConfiguration(
+                    partialPivotConfiguration,
+                    metricQuery,
+                ),
+            };
+
+            return pivotConfiguration;
+        }
+    }
+
+    return undefined;
+}
+
+export function derivePivotConfigurationFromChart(
+    savedChart: SavedChartDAO,
```

Comment:
We could make it more specific `Pick<SavedChartDAO, 'chartConfig' | 'pivotConfig'>`. Helps with usage and test mocking.